community/kernel/src/main/java/org/neo4j/kernel/recovery/LatestCheckPointFinder.java

```diff
@@ -91,9 +93,28 @@ public LatestCheckPoint find( long fromVersionBackwards ) throws IOException
 
             if ( latestCheckPoint != null )
             {
+                // Is the latest start entry in this log file version later than what the latest check point targets?
+                LogPosition target = latestCheckPoint.getLogPosition();
                 boolean commitsAfterCheckPoint = latestStartEntry != null &&
-                        latestStartEntry.getStartPosition().compareTo( latestCheckPoint.getLogPosition() ) >= 0;
-                return new LatestCheckPoint( latestCheckPoint, commitsAfterCheckPoint , oldestVersionFound );
+                        latestStartEntry.getStartPosition().compareTo( target ) >= 0;
+                if ( !commitsAfterCheckPoint )
+                {
+                    if ( target.getLogVersion() < version )
+                    {
+                        // This check point entry targets a previous log file.
+                        // Go there and see if there's a transaction. Reader is capped to that log version.
+                        commitsAfterCheckPoint = extractFirstTxIdAfterPosition( target, target.getLogVersion() ) !=
```

Comment:
I think this code
```
if ( !startEntryAfterCheckPoint )
{
    if ( target.getLogVersion() < version )
    {
        // This check point entry targets a previous log file.
        // Go there and see if there's a transaction. Reader is capped to that log version.
        startEntryAfterCheckPoint = extractFirstTxIdAfterPosition( target, target.getLogVersion() ) !=
                LatestCheckPoint.NO_TRANSACTION_ID;
    }
}
```
when `extractFirstTxIdAfterPosition` will start at log version of `target` and then stop at `target.getLogVersion()` which means it will not search any intermediate log versions between `target.getLogVersion` and wherever we found the checkpoint.

To be clear
```
--- Log version 1
tx1
--- Log version 2
tx2
tx3
...
--- Log version 3
Checkpoint targeting tx 1
```
Here, `target` is `tx1` which live in log version 1. This is where we will start our search when trying to `extractFirstTxIdAfterPosition`. What we want to find is `tx2` which is the first tx id after position, right? Oooor, is our `target` position the start of `tx1` and what we want to find is the commit entry of `tx1`? If so, is it safe for us to assume start and commit of `tx1` is in the same log version?